pkg/services/ssosettings/api/api_test.go

```diff
@@ -367,6 +368,174 @@ func TestSSOSettingsAPI_GetForProvider(t *testing.T) {
 	}
 }
 
+func TestSSOSettingsAPI_List(t *testing.T) {
+	type TestCase struct {
+		desc                string
+		action              string
+		scope               string
+		expectedResult      []*models.SSOSettings
+		expectedError       error
+		expectedServiceCall bool
+		expectedStatusCode  int
+	}
+
+	tests := []TestCase{
+		{
+			desc:   "successfully lists SSO settings",
+			action: "settings:read",
+			scope:  "settings:auth.azuread:*",
+			expectedResult: []*models.SSOSettings{
+				{
+					ID:        "1",
+					Provider:  "azuread",
+					Settings:  make(map[string]interface{}),
+					Created:   time.Now(),
+					Updated:   time.Now(),
+					IsDeleted: false,
+					Source:    models.DB,
+				},
+				{
+					ID:        "2",
+					Provider:  "github",
+					Settings:  make(map[string]interface{}),
+					Created:   time.Now(),
+					Updated:   time.Now(),
+					IsDeleted: false,
+					Source:    models.DB,
+				},
```

Comment:
Something feels wrong here. The scope is defined as `settings:auth.azuread:*` but the expected result contains the `github` provider as well. Is this expected?

Reply:
Agree, I left some comments in the file on some test cases below, but I don't think these tests are filtering results based on auth properly right now. I'm still looking into why.